lib/pdf/conversion.rb

```diff
@@ -23,13 +23,25 @@ def self.generate_from_html(html, outpath, options = {})
 
   PDF_GENERATION_TIMEOUT = 15 * 60
 
-  def self.invoke_generation_script(args, options = {})
+  def self.invoke_generation_script(args, options = {}, retry_attempts: 3)
     script_path = "#{deploy_dir}/bin/generate-pdf"
     cmd = (['timeout', PDF_GENERATION_TIMEOUT.to_s, 'node', script_path] + args).join(" ")
     puts cmd if options[:verbose]
     @result = `#{cmd}`
     if $?.exitstatus == 124
       raise "pdf generation timed out after #{PDF_GENERATION_TIMEOUT} seconds. cmd: #{cmd}"
+    elsif $?.exitstatus == 1
+      # We have had inconsistent issues with pdf generation timing out thus here 
+      # we are re-attempting generation using recursion.  We considered using a loop
+      # instead of reucursion, but the code was less readable or had side effects that 
+      # were not addressing this specific issue with time out. We do not see an issue 
+      # with performance in this case.
```

Comment:
Super-tiny nit; what about something like,

```suggestion
      # were not addressing this specific issue with time out. Because this method is
      # invoked as part of the staging build rather than anything user-facing, we do
      # not see an issue with performance in this case.
```